File: code/cpp/src/topic_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>  // <1>


int main(int argc, char **argv) {
  ros::init(argc, argv, "count_publisher");
  ros::NodeHandle node;

  ros::Publisher pub = node.advertise<std_msgs::Int32>("counter", 10); // <2>

  ros::Rate rate(1);  // <3>
  int count = 0;

  while (ros::ok()) {  // <4>
    std_msgs::Int32 msg; // <5>
    msg.data = count;

    pub.publish(msg);  // <6>

    ++count;
    rate.sleep();  // <7>
  }

  return 0;  // <8>
}

File: code/basics/src/topic_publisher.py
#!/usr/bin/env python
# BEGIN ALL

# BEGIN IMPORT
import rospy
# END IMPORT

# BEGIN STD_MSGS
from std_msgs.msg import Int32
# END STD_MSGS


rospy.init_node('topic_publisher')

# BEGIN PUB
pub = rospy.Publisher('counter', Int32)
# END PUB

# BEGIN LOOP
rate = rospy.Rate(2)

count = 0
while not rospy.is_shutdown():
    pub.publish(count)
    count += 1
    rate.sleep()
# END LOOP
# END ALL
